src/types/iterators.jl

```diff
@@ -0,0 +1,31 @@
+import Base.length, Base.getindex, Base.iterate
+
+function length(o::GBIFRecords)
+  length(o.occurrences)
+end
+
+function iterate(o::GBIFRecords)
```

Comment:
couldn't you replace this entire method by just defining the next method as `function iterate(o::GBIFRecords, state::Int64 = findfirst(o.show))`?